tests/phpunit/tests/filesystem/moveDir.php

```diff
@@ -0,0 +1,285 @@
+			$this->assertTrue(
+				$wp_filesystem->exists( $from ),
+				'The $from directory does not exist anymore.'
+			);
+
+			if ( false === $overwrite && 'existing_to' === untrailingslashit( $to ) ) {
+				$this->assertTrue(
+					$wp_filesystem->exists( $to ),
+					'The $to directory does not exist anymore.'
+				);
+			}
+		}
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_should_return_wp_error() {
+		return array(
+			'$overwrite is false and $to exists' => array(
+				'from'      => 'existing_from',
+				'to'        => 'existing_to',
+				'overwrite' => false,
+				'expected'  => 'destination_already_exists_move_dir',
+			),
+			'same source and destination, source has trailing slash' => array(
+				'from'      => 'existing_from/',
+				'to'        => 'existing_from',
+				'overwrite' => false,
+				'expected'  => 'source_destination_same_move_dir',
+			),
+			'same source and destination, destination has trailing slash' => array(
+				'from'      => 'existing_from',
+				'to'        => 'existing_from/',
+				'overwrite' => false,
+				'expected'  => 'source_destination_same_move_dir',
+			),
+			'same source and destination, source lowercase, destination uppercase' => array(
+				'from'      => 'existing_from',
+				'to'        => 'EXISTING_FROM',
+				'overwrite' => false,
+				'expected'  => 'source_destination_same_move_dir',
+			),
+			'same source and destination, source uppercase, destination lowercase' => array(
+				'from'      => 'EXISTING_FROM',
+				'to'        => 'existing_from',
+				'overwrite' => false,
+				'expected'  => 'source_destination_same_move_dir',
+			),
+			'same source and destination, source and destination in inverted case' => array(
+				'from'      => 'ExIsTiNg_FrOm',
+				'to'        => 'eXiStInG_fRoM',
+				'overwrite' => false,
+				'expected'  => 'source_destination_same_move_dir',
+			),
+		);
+	}
+
+	/**
+	 * Tests that move_dir() successfully moves a directory.
+	 *
+	 * @ticket 57375
+	 *
+	 * @dataProvider data_should_move_directory
+	 *
+	 * @param string $from      The source directory path.
+	 * @param string $to        The destination directory path.
+	 * @param bool   $overwrite Whether to overwrite the destination directory.
+	 */
+	public function test_should_move_directory( $from, $to, $overwrite ) {
+		global $wp_filesystem;
+
+		$from   = self::$test_dir . $from;
+		$to     = self::$test_dir . $to;
+		$result = move_dir( $from, $to, $overwrite );
+
+		$this->assertTrue(
+			$result,
+			'The directory was not moved.'
+		);
+
+		$this->assertFalse(
+			$wp_filesystem->exists( $from ),
+			'The source directory still exists.'
+		);
+
+		$this->assertTrue(
+			$wp_filesystem->exists( $to ),
+			'The destination directory does not exist.'
+		);
+
+		$dirlist = $wp_filesystem->dirlist( $to, true, true );
+
+		$this->assertIsArray(
+			$dirlist,
+			'The directory listing of the destination directory could not be retrieved.'
+		);
+
```

Comment:
Can be removed as `assertSameSets` includes the assertion.

Reply:
Updated in [d959eb9](https://github.com/WordPress/wordpress-develop/pull/3909/commits/d959eb9568e3e2a2a0a2109a5719ee46c8fc6f3f).